customer-endpoint/src/main/kotlin/org/ostelco/prime/customer/endpoint/resources/SimProfilesResource.kt

```diff
@@ -68,6 +68,24 @@ class SimProfilesResource(private val regionCode: String, private val dao: Subsc
                         .responseBuilder()
             }.build()
 
+    @PUT
+    @Path("/{iccId}/installed")
+    @Produces(MediaType.APPLICATION_JSON)
+    fun updateSimProfile(@NotNull
+                         @PathParam("iccId")
+                         iccId: String,
+                         @Auth
+                         token: AccessTokenPrincipal?): Response =
```

Comment:
It is a good idea to check if the incoming parameter is actually an ICCID. Check if it's a number sequence is of the correct length (19-21 characters, only digits (decimal)).

Reply:
There are cases such as physical SIM cards where we do not have real ICCID.  We then use UUID with some special prefix to indicate that.  So, we cannot use the validation proposed above.